### ISSUE TITLE
Unable to resolve resourceDuration metric

### ISSUE THREAD

**Issue by lyonsaj** [argoproj/argo#3292]
Checklist:

 * [ ] I've included the version.
 * [ ] I've included reproduction steps.
 * [ ] I've included the workflow YAML.
 * [ ] I've included the logs.
 
**What happened**:

resourceDuration metric does not resolve.
resourceDuration.cpu and resourceDuration.memory do not resolve either 

**What you expected to happen**:

A metric being emitted with the resource duration.
**How to reproduce it (as minimally and precisely as possible)**:

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: retry-to-completion-
spec:
  entrypoint: retry-to-completion
  metrics:
    prometheus:
      - name: result_counter
        help: "Count of step execution by result status"
        labels:
          - key: name
            value: "retry"
          - key: duration
            value: "{{workflow.duration}}"
          - key: resource_duration_cpu
            value: "{{resourceDuration.cpu}}"
        counter:
          value: "1"
  templates:
  - name: retry-to-completion
    retryStrategy: {}
    container:
      image: python
      command: ["python", "-c"]
      # fail with a 80% probability
      args: ["import random; import sys; exit_code = random.choice(range(0, 5)); sys.exit(exit_code)"]
```
**Anything else we need to know?**:

**Environment**:

- Argo version:
```
$ argo version
argo: v2.7.6+70facdb.dirty
  BuildDate: 2020-04-28T17:09:06Z
  GitCommit: 70facdb67207dbe115a9029e365f8e974e6156bc
  GitTreeState: dirty
  GitTag: v2.7.6
  GoVersion: go1.13.4
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: "2020-01-15T08:26:26Z"
  compiler: gc
  gitCommit: e7f962ba86f4ce7033828210ca3556393c377bcc
  gitTreeState: clean
  gitVersion: v1.16.6-beta.0
  goVersion: go1.13.5
  major: "1"
  minor: 16+
  platform: darwin/amd64
serverVersion:
  buildDate: "2020-02-13T23:24:07Z"
  compiler: gc
  gitCommit: 7be6fd53a1c9885bf9f470673f1c14280aab6386
  gitTreeState: clean
  gitVersion: v1.16.7+vmware.1
  goVersion: go1.13.6
  major: "1"
  minor: "16"
  platform: linux/amd64

```

**Other debugging information (if applicable)**:

- workflow result:

```
argo --loglevel DEBUG get <workflowname>

Name:         retry-to-completion-j4c2l
Namespace:    argo
Labels:       workflows.argoproj.io/completed=true
              workflows.argoproj.io/phase=Succeeded
Annotations:  <none>
API Version:  argoproj.io/v1alpha1
Kind:         Workflow
Metadata:
  Creation Timestamp:  2020-06-24T19:04:41Z
  Generate Name:       retry-to-completion-
  Generation:          8
  Resource Version:    11961434
  Self Link:           /apis/argoproj.io/v1alpha1/namespaces/argo/workflows/retry-to-completion-j4c2l
  UID:                 0702725b-e6b4-4a84-9855-377d21a86347
Spec:
  Arguments:
  Entrypoint:  retry-to-completion
  Metrics:
    Prometheus:
      Counter:
        Value:    1
      Gauge:      <nil>
      Help:       Count of step execution by result status
      Histogram:  <nil>
      Labels:
        Key:             name
        Value:           retry
        Key:             duration
        Value:           {{workflow.duration}}
        Key:             resource_duration_cpu
        Value:           {{resourceDuration.cpu}}
      Name:              result_counter
      When:              
  Service Account Name:  argo
  Templates:
    Arguments:
    Container:
      Args:
        import random; import sys; exit_code = random.choice(range(0, 5)); sys.exit(exit_code)
      Command:
        python
        -c
      Image:  python
      Name:   
      Resources:
    Inputs:
    Metadata:
    Name:  retry-to-completion
    Outputs:
    Retry Strategy:
Status:
  Conditions:
    Status:     True
    Type:       Completed
    Message:    unable to substitute parameters for metric 'result_counter': failed to resolve {{resourceDuration.cpu}}
    Status:     True
    Type:       MetricsError
  Finished At:  2020-06-24T19:04:49Z
  Nodes:
    retry-to-completion-j4c2l:
      Children:
        retry-to-completion-j4c2l-717780961
        retry-to-completion-j4c2l-248154724
      Display Name:  retry-to-completion-j4c2l
      Finished At:   2020-06-24T19:04:49Z
      Id:            retry-to-completion-j4c2l
      Name:          retry-to-completion-j4c2l
      Outputs:
        Exit Code:     0
      Phase:           Succeeded
      Started At:      2020-06-24T19:04:41Z
      Template Name:   retry-to-completion
      Template Scope:  local/retry-to-completion-j4c2l
      Type:            Retry
    retry-to-completion-j4c2l-248154724:
      Display Name:    retry-to-completion-j4c2l(1)
      Finished At:     2020-06-24T19:04:48Z
      Host Node Name:  ip-10-105-8-72.ec2.internal
      Id:              retry-to-completion-j4c2l-248154724
      Name:            retry-to-completion-j4c2l(1)
      Outputs:
        Exit Code:     0
      Phase:           Succeeded
      Started At:      2020-06-24T19:04:45Z
      Template Name:   retry-to-completion
      Template Scope:  local/retry-to-completion-j4c2l
      Type:            Pod
    retry-to-completion-j4c2l-717780961:
      Display Name:    retry-to-completion-j4c2l(0)
      Finished At:     2020-06-24T19:04:44Z
      Host Node Name:  ip-10-105-8-72.ec2.internal
      Id:              retry-to-completion-j4c2l-717780961
      Message:         failed with exit code 4
      Name:            retry-to-completion-j4c2l(0)
      Outputs:
        Exit Code:     4
      Phase:           Failed
      Started At:      2020-06-24T19:04:41Z
      Template Name:   retry-to-completion
      Template Scope:  local/retry-to-completion-j4c2l
      Type:            Pod
  Phase:               Succeeded
  Started At:          2020-06-24T19:04:41Z
Events:
  Type     Reason                 Age   From                       Message
  ----     ------                 ----  ----                       -------
  Normal   WorkflowRunning        83s   retry-to-completion-j4c2l  Workflow Running
  Warning  WorkflowNodeFailed     79s   retry-to-completion-j4c2l  Failed node retry-to-completion-j4c2l(0): failed with exit code 4
  Warning  WorkflowNodeFailed     78s   retry-to-completion-j4c2l  Failed node retry-to-completion-j4c2l(0): failed with exit code 4
  Normal   WorkflowNodeSucceeded  75s   retry-to-completion-j4c2l  Succeeded node retry-to-completion-j4c2l(1)
  Normal   WorkflowNodeSucceeded  75s   retry-to-completion-j4c2l  Succeeded node retry-to-completion-j4c2l
  Normal   WorkflowSucceeded      75s   retry-to-completion-j4c2l  Workflow completed

```

---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by simster7** [comment]
I am almost certain that this metric is not available in `2.7.6`. Upgrade to `2.8.2`?

**Comment by lyonsaj** [comment]
@simster7 I'm also seeing it with workflow.failures. I have upgraded back to 2.8.2 and still get it.

**Comment by simster7** [comment]
Hey @lyonsaj sorry, I edited my comment to say that this metric is only available in 2.9+. The fix for workflow.failures also won't be available till 2.9

**Comment by firecast** [comment]
@simster7 Facing the same issue on argo version `3.2.3` as well. Error is the same.
```
 MetricsError: unable to substitute parameters for metric 'result_counter': failed to resolve {{resourceDuration.cpu}}
```
